src/scheduler/queue_info.c

```diff
@@ -666,7 +666,7 @@ update_queue_on_run(queue_info *qinfo, resource_resv *resresv, char *job_state)
 			multi_node_sort);
 
 
-	if ((job_state != NULL) && (*job_state == 'S'))
+	if ((job_state != NULL) && (*job_state == 'S') && (resresv->job->resreq_rel != NULL))
```

Comment:
This is a side effect of assuming preemption method "S" internally in sched when calling update_universe_on_end() for any preemption method. Ideally, job_state wouldn't be "S" for preempt method 'R' right? I'd wanted to change this in the past, but I think it wasn't worth the effort at the time, so I just wanted to check, is it worth fixing it now? We might be able to avoid such bugs in the future if we fix this.

Reply:
Actually now that i look at the code, i see that update_universe_on_end() is in fact not called with job state "S" all the time. So, why is the job_state 'S' for preempt method R  ?

Reply:
We only call update_universe_on_end() with job state S inside the simulation.  Outside the simulation after we have preempted jobs, we correctly call it with job state Q.  Inside the simulation, it doesn't matter what we call it with because we won't look at the job again.  This isn't really an additional bug to what you fixed, but an extension of it.  What you fixed was this bug for nodes.  We just didn't realize it was also broken for queues and the server back then.  I think Vincent's current fix is fine.